DEVELOPER_GUIDE.md

```diff
@@ -134,6 +135,17 @@ To build a distribution to run on your local platform, run:
 
 All distributions built will be under `distributions/archives`.
 
+#### Generated Code
+
+OpenSearch uses code generators like [Protobuf](https://protobuf.dev/).
+OpenSearch build system already takes a dependency of generating code from protobuf, incase you run into compilation errors, run:
```

Comment:
In which case will we run into compilation errors? 
Also, since `generateProto` is already included in srever/build.gradle we shouldn't run the below again, right?

Reply:
Usual workflows shouldn't run into any errors. 
@kotwanikunal already built a workspace on local and while running tests he did a rebase which ran into compilation problems. The rebase had new proto files and java code which had dependencies on that. 

Instead of running the whole `compileJava` this is just lighter. See: https://github.com/opensearch-project/OpenSearch/pull/7302#issuecomment-1533416296

Anyway this is just incase if somebody is hitting into problems, we dont expect any with usual workflows.